tasks/linux.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+action=$PT_action
+name=$PT_name
+command_exists=0
+service_managers[0]="systemctl"
+service_managers[1]="service"
+service_managers[2]="initctl"
+
+# example cli /opt/puppetlabs/puppet/bin/bolt  task run service::linux action=stop name=ntp --nodes localhost --modulepath /etc/puppetlabs/code/modules --password puppet --user root
+
+check_command_exists() {
+  (which $1) > /dev/null 2>&1
+  command_exists=$?
+}
+
+for service_manager in "${service_managers[@]}"
+do
+  check_command_exists "$service_manager"
+  if [ $command_exists -eq 0 ]; then
+    command_line="$service_manager $action $name"
+    if [ $service_manager == "service" ]; then
+      command_line="$service_manager $name $action"
+    fi
+    output=$($command_line 2>&1)
+    status_from_command=$?
+    # set up our status and exit code
+    if [ $status_from_command -eq 0 ]; then
+      echo "{ \"status\": \"$name $action\" }"
+      exit 0
+    else
+      # initd is special, starting an already started service is an error
+      if [[ $service_manager == "service" && $output == *"Job is already running"* ]]; then
```

Comment:
`$output` needs quotes here; it might be interesting to people to also see the output in the json result?